### PR TITLE
[stdlib] Remove outdated miscellaneous functions

### PR DESCRIPTION
**Pull request by xwu**
Now that we have protocol-based integers, the "miscellaneous" free functions are no longer necessary to compute leading zeros and binary logarithms. So let's remove them in favor of the corresponding modern facilities.

### CODE REVIEW

**Comment by xwu** [comment]
@swift-ci Please smoke test

**Comment by xwu** [comment]
Test failure is unrelated and addressed in apple/swift-lldb#300.

**Comment by xwu** [comment]
@swift-ci Please smoke test OS X platform

**Review by airspeedswift — APPROVED** [review]
LGTM.

File: stdlib/public/core/Misc.swift [hunk 1]
```diff
@@ -14,13 +14,6 @@
 
 // FIXME: Once we have an FFI interface, make these have proper function bodies
 
-@_inlineable // FIXME(sil-serialize-all)
-@_transparent
-public // @testable
-func _countLeadingZeros(_ value: Int64) -> Int64 {
```
Comment (moiseev) [inline stdlib/public/core/Misc.swift]: Wow. I did not know we had this!

Comment (xwu) [reply stdlib/public/core/Misc.swift]: More than five years old!

**Pull request merged**